<launch>
    <arg name="pkg" default="$(find semantic_bki)" />
    <arg name="method" default="semantic_bki"/>
    <!-- <arg name="method" default="semantic_csm"/> -->
    <arg name="dataset" default="senior"/>
   
    <arg name="method_config" value="$(arg pkg)/config/methods" />
    <arg name="data_config" value="$(arg pkg)/config/datasets/" />
    <arg name="matrix_config" value="$(arg pkg)/config/matrix/" />
    

    <!-- <node type="rviz" name="rviz" pkg="rviz" args="-d $(find semantic_bki)/rviz/multi_node.rviz" /> -->

    <node pkg="semantic_bki" type="multi_senior_node" name="robot1"   output="screen">
        <!-- Load method configuration from file -->
        <rosparam file="$(arg method_config)/$(arg method).yaml" command="load" />
        <!-- Load dataset configuration from file -->
        <rosparam file="$(arg data_config)/$(arg dataset).yaml" command="load" />    
        <rosparam file="$(arg matrix_config)/r2tor1.yaml" command="load" />    
        <param name="name" type="string" value="/robot1" />
        <param name="frientname" type="string" value="/robot2" />
        <param name="cloudTP" type="string" value="/seg_cloud" />
    </node>

    <node pkg="rosbag" type="play" name="rosbag1" 
		args="/home/dengyinnan/dataset/robot3_raw.bag -r 0.25"/>

    <node pkg="semantic_bki" type="multi_senior_node" name="robot2"  output="screen">
        <!-- Load method configuration from file -->
        <rosparam file="$(arg method_config)/$(arg method).yaml" command="load" />
        <!-- Load dataset configuration from file -->
        <rosparam file="$(arg data_config)/$(arg dataset).yaml" command="load" />    
        <rosparam file="$(arg matrix_config)/r1tor2.yaml" command="load" />    
        <param name="name" type="string" value="/robot2" />
        <param name="frientname" type="string" value="/robot1" />
        <param name="cloudTP" value="/seg_cloud_2" />
		<remap from="/tf" to="/tf2" />
    </node>
    
    <node pkg="rosbag" type="play" name="rosbag2" 
		args="/home/dengyinnan/dataset/robot4_raw.bag -r 0.25  /seg_cloud:=/seg_cloud_2 /tf:=/tf2 
                  /camera/right/image_raw/compressed:=/camera/right/image_raw/compressed2  velodyne_points:=velodyne_points2"/>
    


</launch>
